mozilla/contain-facebook
# Refactor content panels into an iframe

This PR refactors the content panels into an iframe, instead of injecting it directly into the DOM.


Preview:
<img width="513" alt="image" src="https://user-images.githubusercontent.com/13066134/176215826-f7bb00d8-0197-475e-9778-1a67c6e5fd68.png">

<img width="542" alt="image" src="https://user-images.githubusercontent.com/13066134/176215903-7d0d82aa-86c5-4404-bcda-ec7d70c45756.png">

UI is updated to reflect new [Firefox prompt designs](https://www.figma.com/file/2ruSnPauajQGprFy6K333u/Desktop-Components?node-id=303%3A9478).

Post-review To-Dos:
- [x] Add checkbox 'do not show again' + escape key functionality 
- [x] Change `innerHTML` to `textContent`
- [x] UI Redesign to follow Firefox popup design specs
- [x] Update cross-origin communication logic

Remaining To-Dos:

- [x] Position `iframe` window according to FBC icon position
- [x] Localize strings
- [x] Make `iframe` position responsive to screen resize/scroll
- [x] Close popup when selecting outside `iframe` window and when _cancel_ + _dismiss_ is selected
- [x] Add Facebook authentication backlink to _Allow_ button
- [x] Add left/right chevron to `iframe` window
- [x] Restore `allowClickSwitch` boolean to `login` switch case
- [x] Mobile orientation

### Testing
Case 1 (Login popup)
1. Go to [Imgur sign in page](https://imgur.com/signin)
2. Select the FBC Icon and check that the prompt opens 
3. Close the popup by clicking Dismiss or selecting outside of the `iframe` window
5. Open the popup again and select the _Allow_ CTA
6. Check that window is redirected to a Facebook authentication page

Case 2 (Login + Email popup) Note: Popups should contain different content
1. Go to [Etsy sign in page](https://www.etsy.com/ca/signin)
2. Select the FBC icon at the top over the email input field and check that the prompt opens + contains a checkbox and a _Try Firefox Relay_ CTA
3. Close the popup by clicking Dismiss or selecting outside of the `iframe` window
4. Select FBC icon at the bottom over the disabled _Continue with Facebook_ button and check that the prompt opens + contains an _Allow_ CTA
7. Close the popup by clicking Dismiss or selecting outside of the `iframe`  window
